src/Compilers/Core/Portable/WellKnownMembers.cs

```diff
@@ -992,6 +992,17 @@ static WellKnownMembers()
                 0,                                                                                                          // Arity
                     (byte)SignatureTypeCode.GenericTypeParameter, 0,                                                        // Field Signature
 
+                // System_Runtime_CompilerServices_RuntimeHelpers__CreateSpanRuntimeFieldHandle
```

Comment:
The doc for new `CreateSpan` API should mention `uint` and probably also `bool` (although the compiler doesn't use it for that case)

Reply:
What about `nint` and `nuint`? Can this optimization not work for that as well?

Reply:
Those already aren't supported for arrays, e.g. RuntimeHelpers.InitializeArray.  It'd be hard to do that efficiently... the data is baked into the assembly, but their size can change based on that of the process into which the assembly is loaded.  So you'd have to pick one size and then at runtime adapt if it wasn't that.  We do the same for endianness, but big endian machines are rare (and aren't even supported with coreclr); 32-bit vs 64-bit is much more of a coin flip.

Reply:
Chatted with @davidwrighton and there's a few things we need to follow up on regarding field RVA:

1. Is the compiler emitting properly aligned blobs?
David thinks the compiler always produces blobs aligned with 8 bytes (ie. a bit inefficient, but safe).
We'll need to double-check that, and fix if not the case.

2. System.Reflection.Metadata
This assembly is used by the compiler to emit PE. But it used to have a bug for scenarios with blobs aligned with 8 bytes.
This was fixed in January (PR https://github.com/dotnet/runtime/pull/63305), but we need to make sure that roslyn is using an up-to-date version of this binary, which is apparently a bit complicated. I will follow-up with Jared on Monday.

3. IL rewriters 
IL rewriters need to maintain alignment. We're ignoring CCI, so that leave Cecil scenarios.
Cecil was updated (PR https://github.com/mono/cecil/pull/33) to maintain alignment, but that design relies on the compiler tagging the field with a "pack" value (a number, such as 1, 2, 4, 8, ...., part of the field layout details).
We'll want the compiler to emit the "pack" value.
The trimmer uses Cecil and is part of the SDK, so users will automatically have this fix as part of their .NET 7 build.
But any customers that have custom trimmers based on Cecil would run into issues if their Cecil is not up-to-date. We could add some documentation in breaking changes [doc](https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Compiler%20Breaking%20Changes%20-%20DotNet%207.md) as mitigation.
In terms of failure mode, the `CreateSpan` API currently detects that the blob is not properly aligned and will explicitly fail.

Reply:
>  we need to make sure that roslyn is using an up-to-date version of this binary, which is apparently a bit complicated. 

Yeah that will be a bit complicated. The fix went into `main` which means it will appear in the 7.0.0.0 version of SRM. That releases as a part of `net7.0` but we need it in the compiler, at RTM versions, before RTM date. Updating SRM or SCI is always "fun" as we have to change the VS versions and that is a bit of a hassle. 

Is there any chance of this being back ported to 6.0.0.0 servicing? That is the easiest route as it gives us the most wiggle room for getting everything inserted.

Reply:
@jaredpar It looks like the SRM fix was backported to .NET 6 (PR https://github.com/dotnet/runtime/pull/64559) :-)

Reply:
> We'll want the compiler to emit the "pack" value.

I've pushed a commit that does so.  Let me know if it's what you had in mind.

Reply:
I added a src assert and test for (1) as well.

Reply:
@stephentoub Quick update. From chat with Jared, we're going to wait for .NET 7 to RTM, then it should be relatively easy to insert RTM bits of SRM into VS.
I'm going to mark the PR as draft for now to make room in our review queue. Can you re-activate and ping after .NET 7 RTMs?

Reply:
It looks like the work to insert newer SRM bits into VS is under way, although it might be two steps (6.0.1 first then 7). 6.0.1 would be sufficient for this PR. Here's the [VS PR to track](https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/434270).